python/mlc_chat/quantization/ft_quantization.py

```diff
@@ -117,11 +123,30 @@ def visit_module(self, name: str, node: nn.Module) -> Any:
                 ret_node: Any
                     The new node to replace current node.
                 """
-                if isinstance(node, nn.Linear) and node.out_features != "vocab_size":
-                    # This skips lm_head in llama; otherwise the quantized shape would be an op
-                    # See https://github.com/mlc-ai/mlc-llm/issues/1723
+                if isinstance(node, nn.Linear):
                     weight_name = f"{name}.weight"
                     self.quant_map.param_map[weight_name] = [f"{name}.q_weight", f"{name}.q_scale"]
+                    if (
+                        # pylint: disable=too-many-boolean-expressions
+                        is_final_fc(name)
+                        or node.weight.dtype == "float32"
```

Comment:
why do we check the weight dtype? previous implementation checks the output dtype is not float32 because mixed precision (fp16 weight, fp32 output) is not supported

Reply:
I see; so this should be `node.out_dtype == "float32"` instead right?

Reply:
right